### PR TITLE
GEODE-8076: simplify redis concurrency code

### PR DESCRIPTION
**Pull request by prettyClouds**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/rest/internal/web/controllers/PutKeyFunction.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
 
   private static final String ID = "PutKeyFunction";
 
+  @SuppressWarnings("unchecked")
   @Override
```
Comment (jdeppe-pivotal) [inline geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/rest/internal/web/controllers/PutKeyFunction.java:34]: Is it possible to add correct generic types here so we don't need to suppress anything?

File: geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/rest/internal/web/controllers/GetAllEntries.java [hunk 2]
```diff
@@ -29,6 +29,7 @@
 
 public class GetAllEntries implements Function {
 
+  @SuppressWarnings("unchecked")
```
Comment (jdeppe-pivotal) [inline geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/rest/internal/web/controllers/GetAllEntries.java:32]: Is it possible to add correct generic types here so we don't need to suppress anything?

File: geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/rest/internal/web/controllers/PutKeyFunction.java (hunk 1 above)
Comment (dschneider-pivotal) [reply geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/rest/internal/web/controllers/PutKeyFunction.java]: I have removed this change. I don't know how it snuck in. It caused a bunch of merge conflicts. The build is fine without this change. If someone wants to clean this test code up I think it should be on another PR.

File: geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/rest/internal/web/controllers/GetAllEntries.java (hunk 2 above)
Comment (dschneider-pivotal) [reply geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/rest/internal/web/controllers/GetAllEntries.java]: I have removed this change. I don't know how it snuck in. It caused a bunch of merge conflicts. The build is fine without this change. If someone wants to clean this test code up I think it should be on another PR.

**Pull request merged**